sound/soc/sof/intel/cnl.c

```diff
@@ -235,12 +235,26 @@ void cnl_ipc_dump(struct snd_sof_dev *sdev)
 
 static int cnl_register_clients(struct snd_sof_dev *sdev)
 {
-	return intel_register_ipc_test_clients(sdev);
+	int ret;
+
+	ret = intel_register_ipc_test_clients(sdev);
+	if (ret < 0)
+		return ret;
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_PROBES)
+	return sof_client_dev_register(sdev, "probes", 0);
+#endif
+
+	return 0;
 }
 
 static void cnl_unregister_clients(struct snd_sof_dev *sdev)
 {
 	intel_unregister_ipc_test_clients(sdev);
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_PROBES)
+	sof_client_dev_unregister(sdev, "probes", 0);
+#endif
```

Comment:
can we add the probes for all of APL/CNL/TGL?

I just tried on TGL and nothing. We should add this for all cAVS1.5+ devices. Thanks!

Reply:
@plbossart unfortunately Dave is having conflicts with TGL on Linus' tree. Thats why I limited the series to just CNL for now

Reply:
APL should not have any conflicts though?

Reply:
but the probes functionality was not validated with the APL firmware.

Reply:
that makes no sense to me. Let's test it then. This only depends on the HDaudio DMA...

Reply:
@fredoh9 is testing it as we speak. All my APL devices gave up on me

Reply:
@plbossart I was under impression that the Probes feature is supported from CNL onward.
I built with APL FW with the probes and I'm testing on UP2. Everything seems to be working fine, except crecord's output file is not growing. I set multiple extraction points and it ends up with setting all of buffer widget IDs but no good news. I need to check with FW team for some clarification. I will update again.

Reply:
Probes is tested in APL UP2. I missed `aplay` when I tested in APL.
Below the diff.txt from @plbossart, it looks great.